### PR TITLE
LPS-145627 Unify Status property style in Remote Apps

### PR DESCRIPTION
**Pull request by javierdearcos**
The Remote Apps table shows the status property with a style that is not consistent with the rest of the portal.

**Steps to reproduce:**

1. Go to the Application Menu > Workflow > Process Builder > Configuration tab and enable the Single Approver workflow for Remote App Entry
2. Go to Application Menu > Remote Apps
3. Create a new remote app
4. Check the style of the pending status label
5. You can also go to the notifications and approved the created remote app
6. Check the style of the approved status label

**Expected result:** The labels are consistent with the portal styles
**Result:** The labels are not consistent with the portal styles
Captura de pantalla 2022-01-19 a las 17 42 37

**Work done:**

I've modified the StatusRenderer from the frontend-dataset-web module to use the same CSS classes and add consistency to the status labels. 

I've also removed the code prop that wasn't used in the renderer and seems to be unnecessary

Result:
Captura de pantalla 2022-01-19 a las 17 35 37

Captura de pantalla 2022-01-19 a las 17 36 00

### CODE REVIEW

**Review by LuismiBarcos — APPROVED** [review]

File: modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/data_renderers/StatusRenderer.js [hunk 1]
```diff
@@ -15,12 +15,33 @@
...
 	return props.value ? (
 		<span className="taglib-workflow-status">
 			<span className="workflow-status">
 				<strong
-					className={`label status workflow-status-data-renderer workflow-status-${props.value.label} ${props.value.label} workflow-value`}
+					className={`label label-${getLabelType(
+						props.value.label
+					)} text-uppercase workflow-status-${props.value.label} ${
+						props.value.label
+					} workflow-value`}
```
Comment (markocikos) [inline modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/data_renderers/StatusRenderer.js:44]: For complex CSS class setting, you can use the `classNames` util. See example [here](https://github.com/liferay/liferay-portal/blob/master/modules/apps/comment/comment-taglib/src/main/resources/META-INF/resources/discussion/js/components/UserIcon.js#L26-L28). Besides being more readable, it allows us to conditionally set some attributes, so we can have something like 
```js
[`label-${labelType}`]: !!labelType
```
In the solution in this PR, if `value.label` is unset or unknown, we would be left with dangling `label-` CSS class.

File: modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/styles/main.scss [hunk 2]
```diff
@@ -1,6 +1,5 @@
 @import './variables';
 
-@import '../data_renderers/data_renderers';
```
Comment (markocikos) [inline modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/styles/main.scss:3]: Nice!

File: modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/data_renderers/StatusRenderer.js [hunk 3]
```diff
@@ -15,12 +15,33 @@
 import PropTypes from 'prop-types';
 import React from 'react';
 
+function getLabelType(label) {
```
Comment (markocikos) [inline modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/data_renderers/StatusRenderer.js:18]: While this helper function would work perfectly fine as it is, it is better scoping if we place it inside the component.
```js
function StatusRenderer(props) {
    const getLabelType = (label) => {...}

    return props.value ?...
```

If it is in root, it should be either exported, or used by multiple components.

**Review by markocikos — CHANGES_REQUESTED** [review]
@javierdearcos respect for treading the treacherous frontend waters! :bow:  See comments inline.

File: modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/data_renderers/StatusRenderer.js (hunk 3 above)
Comment (javierdearcos) [reply modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/data_renderers/StatusRenderer.js:18]: True. Changed and I'll remember in the future

File: modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/data_renderers/StatusRenderer.js (hunk 1 above)
Comment (javierdearcos) [reply modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/data_renderers/StatusRenderer.js:44]: Understood. A

As I don't completely get classNames usage and I have the option to return the complete label-type class from the helper function and it is very clean, I will do it this way.

```
const getLabelType = label => {
		switch (label) {
			case 'approved':
				return 'label-success';
			case 'denied':
				return 'label-danger';
			case 'expired':
				return 'label-warning';
			case 'draft':
			case 'pending':
			case 'scheduled':
				return 'label-info';
			default:
				return '';
		}
	}
```

Please let me know if this way could cause any harm or be a bad option

File: modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/data_renderers/StatusRenderer.js [hunk 4]
```diff
@@ -15,12 +15,33 @@
...
 function StatusRenderer(props) {
 	return props.value ? (
 		<span className="taglib-workflow-status">
 			<span className="workflow-status">
 				<strong
-					className={`label status workflow-status-data-renderer workflow-status-${props.value.label} ${props.value.label} workflow-value`}
+					className={`label label-${getLabelType(
+						props.value.label
+					)} text-uppercase workflow-status-${props.value.label} ${
+						props.value.label
```
[the remark this answers is not in the document]
Comment (javierdearcos) [reply modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/data_renderers/StatusRenderer.js:43]: Ups 🤦 ... Removed them

**Comment by javierdearcos** [comment]
Thanks for all the feedback! 🤗

**Comment by javierdearcos** [comment]
ci:test:sf

**Comment by javierdearcos** [comment]
ci:test:relevant

**Comment by javierdearcos** [comment]
ci:stop

**Comment by javierdearcos** [comment]
ci:test:sf

**Comment by javierdearcos** [comment]
ci:test:relevant

File: modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/data_renderers/StatusRenderer.js (hunk 1 above)
Comment (markocikos) [reply modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/data_renderers/StatusRenderer.js]: That's a nice change! I prefer it like this, half-built classes are harder to read and hard to find when we mass-replace  :+1: 

There is a dangling space character, but that's not an issue. We can push this as it is.

About `classNames`, if you ever feel adventurous, see good and short tutorial on https://github.com/JedWatson/classnames#usage

**Review by markocikos — APPROVED** [review]

File: modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/data_renderers/StatusRenderer.js (hunk 1 above)
Comment (javierdearcos) [reply modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/data_renderers/StatusRenderer.js]: thank you!

**Comment by javierdearcos** [comment]
ci:forward

**Comment by markocikos** [comment]
ci:forward

**Pull request closed without merging**